publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -96,22 +100,34 @@ public void setRestTemplate(RestTemplate restTemplate) {
             @ApiResponse(code = 503, message = "Service Unavailable") })
     @RequestMapping(value = "/producer/msg", method = RequestMethod.POST)
     @ResponseBody
-    public ResponseEntity send(@ApiParam(value = "message protocol", required = true) @RequestParam(value = "mp") final String msgProtocol,
-                               @ApiParam(value = "user domain") @RequestParam(value = "ud", required = false) final String userDomain,
-                               @ApiParam(value = "tag") @RequestParam(value = "tag", required = false) final String tag,
-                               @ApiParam(value = "routing key") @RequestParam(value = "rk", required = false) final String routingKey,
-                               @ApiParam(value = "eiffel event", required = true) @RequestBody final JsonElement body) {
+    public ResponseEntity send(
+            @ApiParam(value = "message protocol", required = true) @RequestParam(value = "mp") final String msgProtocol,
+            @ApiParam(value = "user domain") @RequestParam(value = "ud", required = false) final String userDomain,
+            @ApiParam(value = "tag") @RequestParam(value = "tag", required = false) final String tag,
+            @ApiParam(value = "routing key") @RequestParam(value = "rk", required = false) final String routingKey,
+            @ApiParam(value = "eiffel event", required = true) @RequestBody final JsonElement body) {
         MsgService msgService = PublishUtils.getMessageService(msgProtocol, msgServices);
         log.debug("mp: " + msgProtocol);
         log.debug("body: " + body);
-        log.debug("user domain suffix: " + userDomain + " tag: " + tag + " Routing Key: " + routingKey);
+        log.debug("user domain suffix: " + userDomain + " tag: " + tag + " Routing Key: "
+                + routingKey);
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        // Check if the user is authenticated
+        if (authentication != null && authentication.isAuthenticated()) {
+            // Get the UserDetails object, which contains user information
+            UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+            // Get the username of the authenticated user
+            String username = userDetails.getUsername();
+            log.info("Authentication Successful for user {} ", username);
```

Comment:
It works fine, but I found 2 issues:
1. I think "User name: {}" is enough. Non-authenticated user cannot get here.
2. The same log should be done for generatreAndPublish(). Please, don't copy the code, create a function instead.

Reply:
Modified the code.